llmfoundry/utils/exceptions.py

```diff
@@ -0,0 +1,215 @@
+# Copyright 2024 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Custom exceptions for the LLMFoundry."""
+from collections.abc import Mapping
+from typing import Dict, List
+
+
+# Finetuning dataloader exceptions
+class MissingHuggingFaceURLSplitError(ValueError):
+    """Error thrown when there's no split used in HF dataset config."""
+
+    def __init__(self) -> None:
+        message = 'When using a HuggingFace dataset from a URL, you must set the ' + \
+                    '`split` key in the dataset config.'
+        super().__init__(message)
+
+
+class NotEnoughDatasetSamplesError(ValueError):
+    """Error thrown when there is not enough data to train a model."""
+
+    def __init__(self, dataset_name: str, split: str,
+                 dataloader_batch_size: int, world_size: int,
+                 full_dataset_size: int, minimum_dataset_size: int) -> None:
+        self.dataset_name = dataset_name
```

Comment:
do these need to be set as attrs for some reason? same throughout

Reply:
yes, will use them to construct the user facing messages

Reply:
Meaning when you intercept with excepthook? Cause here you just pass it along to ValueError

Reply:
When we catch the exception with the mosaic logger, we can pull the attributes out and also save them in the run metadata.

Reply:
ahh ok. got it.